parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
@johndgiese I'm fine with moving `map`s to imperative loops if you feel it's better style. The intention with many of these `map` calls (I made a lot of them throughout the code base) was to cut down on the number of explicit loops. Unfortunately, BeautifulSoup operates only through side effects on the parse tree, so there's not a graceful way of getting around having to perform so many side-effect-generating loops.